### ISSUE TITLE
fix(electron): "Cannot read property '0' of null"

### ISSUE THREAD

**Issue by zendude213** [athensresearch/athens#749]
**Problem**
Can't install latest Mac update. On opening App, I'm prompted to update.. then it tries for a while and after a few seconds, pix below pops up. If I restart and select "cancel" as opposed to "update", things work as normal

**Screenshots/Demo**
![CleanShot 2021-03-05 at 04 46 24](https://user-images.githubusercontent.com/59263556/110098169-e1e94300-7d6d-11eb-9eb5-8bf6dd57aa3a.png)

**Athens Version**
Version 1.0.0-beta.47 (1.0.0-beta.47)

**Comment by shebang42** [comment]
Same issue.  When upgrading from b47 to (presumably) b50 and from b49 to b50.  On both Macos 11.2 and 10.5.7.

Workaround:

1: **Download** Athens from https://athens-apps.s3.us-east-2.amazonaws.com/Athens-1.0.0-beta.50.dmg
2: **Close Athens**
3: **Install** (== copy to /Applications).  Select '**Replace**' with the popup "An item named “Athens.app” already exists in this location. Do you want to replace it with the one you’re moving?".

**Comment by zendude213** [comment]
thanks.. now works!

**Comment by tangjeff0** [comment]
This happened after #707, which updates `electron` and `electron-builder` so we could make the M1 build. cc @sboesen

**Comment by tangjeff0** [comment]
May also be just be same error as #722

**Comment by sboesen** [comment]
@tangjeff0 does look like the same issue as #722 at first glance - helps narrow it down that it appears to happen just upgrading from beta 46 - 47.

**Comment by tangjeff0** [comment]
The difference between this issue and #722 is that this one seems to prevent people from actually updating.

**Comment by sboesen** [comment]
Here is the error I'm seeing in `~/Library/Logs/Athens/main.log` on Mac when trying to update from Athens beta 49 -> 50:

```
[2021-03-05 16:28:35.401] [info] Error in auto-updater. Error: ZIP file not provided: [
  {
    "url": "https://athens-apps.s3.amazonaws.com/Athens-1.0.0-beta.50-arm64.dmg",
    "info": {
      "url": "Athens-1.0.0-beta.50-arm64.dmg",
      "sha512": "vi1mVDiVLGI9iUDnKxfl6h+7EkZ3imDq4QQ1zD9ildH0HOts9hYnEEa+vOR3LwZW+yzaUurFUGRb/JvnntolQA==",
      "size": 117351332
    }
  },
  {
    "url": "https://athens-apps.s3.amazonaws.com/Athens-1.0.0-beta.50.dmg",
    "info": {
      "url": "Athens-1.0.0-beta.50.dmg",
      "sha512": "xFTsZ9MPfL/37mxKb2sWGzUe2iqMFGiZqNvmfqbGS/m1eAZdJtxWDxlhjfgNe+y+0qMIUmet7qr+ZXEFXBtDWQ==",
      "size": 116473861
    }
  },
  {
    "url": "https://athens-apps.s3.amazonaws.com/Athens-1.0.0-beta.50.dmg",
    "info": {
      "url": "Athens-1.0.0-beta.50.dmg",
      "sha512": "xFTsZ9MPfL/37mxKb2sWGzUe2iqMFGiZqNvmfqbGS/m1eAZdJtxWDxlhjfgNe+y+0qMIUmet7qr+ZXEFXBtDWQ==",
      "size": 116473861
    }
  }
]
```
Seems like `electron-update` may be including a zip file for backwards compatibility, see comment on the ElectronMail project [Electron project](https://github.com/vladimiry/ElectronMail/issues/103#issuecomment-460053141) and [related open issue](https://github.com/electron-userland/electron-builder/issues/2199) on `electron-builder` to remove the extra zip file as the update mechanism.

Taking a look at the manifest file on s3 there's a duplicate entry: https://athens-apps.s3.us-east-2.amazonaws.com/beta-mac.yml

```
  - url: Athens-1.0.0-beta.50.dmg
    sha512: xFTsZ9MPfL/37mxKb2sWGzUe2iqMFGiZqNvmfqbGS/m1eAZdJtxWDxlhjfgNe+y+0qMIUmet7qr+ZXEFXBtDWQ==
    size: 116473861
  - url: Athens-1.0.0-beta.50.dmg
    sha512: xFTsZ9MPfL/37mxKb2sWGzUe2iqMFGiZqNvmfqbGS/m1eAZdJtxWDxlhjfgNe+y+0qMIUmet7qr+ZXEFXBtDWQ==
    size: 116473861
```

This might be a bug in the updated version of `electron-builder`, do you have a copy of the `beta-mac.yml` manifest from the previous Athens release? Currently running into an error fetching some electron update when I try to build the DMG locally, but I would compare that manifest to the one generated by the previous version of `electron-builder` Athens was using

**Comment by tangjeff0** [comment]
Actually the last electron PR you and me tag teamed removed the zip target. Didn’t expect that to be problematic!

Sent via Superhuman iOS ( https://sprh.mn/?vip=tangj1122@gmail.com )

**Comment by sboesen** [comment]
Ahh, whoops! Re-adding that zip target should resolve it, then, at least until electron-updater switches to an all-in-one approach.

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 300.0 USD (300.0 USD @ $1.0/USD)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/athensresearch/athens/749/100024993).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/athensresearch/athens/749/100024993).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.com/invite/83BK5z55yB/'>Gitcoin's Discord</a>
* $2,653,623.54 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by sboesen** [comment]
To clarify is this issue fixing the popup (also referenced in open issue #722), reverting that change made in to the Mac deployment targets in `package.json` to fix auto-update on Mac (what the issue was originally about), or both?

FWIW I [have a commit](https://github.com/sboesen/athens/commit/01646e3) to resolve the update issue (hard to test the end to end flow, but it generates the DMG and zip files now and references them in `beta-mac.yml`), happy to create a pull request and update the commit message per Athens' Standard - just let me know which issue to point the commit message to.

**Comment by tangjeff0** [comment]
> To clarify is this issue fixing the popup (also referenced in open issue #722), reverting that change made in to the Mac deployment targets in package.json to fix auto-update on Mac (what the issue was originally about), or both?

Can you rephrase this question? I don't understand it :sweat_smile: 

I think if we don't have anything for `target` it will default to both ZIP and DMG: https://www.electron.build/configuration/mac

**Comment by sboesen** [comment]
> Can you rephrase this question? I don't understand it 😅

Of course! The issue opened by @zendude213 is that the mac update is broken, and they shared a screenshot of an error that initially appears unrelated though it does appear during update: https://github.com/athensresearch/athens/issues/722

While that error pop up is worth fixing, my impression is that's a separate issue which existed before Mac auto update was broken, and that this issue #749 relates to *just* Mac auto update being broken.

> I think if we don't have anything for target it will default to both ZIP and DMG: https://www.electron.build/configuration/mac @sboesen

@tangjeff0 I think you're right, but [currently we specify just DMG](https://github.com/athensresearch/athens/blob/master/package.json#L23-L30) which appears to break the auto update for Mac.

The code change I [linked earlier](https://github.com/sboesen/athens/commit/01646e3) should fix Mac auto update, if I'm understanding things correctly.

**Comment by tangjeff0** [comment]
Sounds good - open the PR and mention all the relevant issues. We can close them after testing to make sure it works.

**Comment by sboesen** [comment]
Opened PR, #777

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 265 years, 8 months from now.
Please review their action plans below:


**1) [sboesen](https://gitcoin.co/sboesen) has started work.**

Fixed in PR https://github.com/athensresearch/athens/pull/777

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/athensresearch/athens/749/100024993).

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work for 300.0 USD (300.0 USD @ $1.0/USD) has been submitted by__:

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/athensresearch/athens/749/100024993)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/athensresearch/athens/749/100024993).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.com/invite/83BK5z55yB/'>Gitcoin's Discord</a>
* $2,817,365.45 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by sboesen** [comment]
This issue should be reopened, I suspect auto-update now works for M1 macs but after testing from Athens Beta 52 on an Intel mac it downloads the ARM64 update of Beta 53 by default, breaking the current Athens installation.

Current workaround for Intel Macs is to click "Cancel" during update prompt and wait for fix.

Screen Shot 2021-03-09 at 1 53 58 PM

I suspect this may be a bug/feature gap in electron-updater after reviewing the relevant code introduced in the most recent electron-builder release 22.10.5 in an attempt to fix auto-update for M1 macs (code here, relevant PR describing the issue were trying to fix) and I'm still looking to see if there's something we can do to avoid this issue or if we need to roll back electron-builder to 22.10.4 (which doesn't include these filters).

Will be testing on my own S3 bucket to see if I can get this working, should have an update later today or tomorrow morning. We'll definitely want to make sure that this works on M1 macs too after the change (and doesn't download the Intel version), something I'm unfortunately still limited in testing.

**Comment by tangjeff0** [comment]
That link was wrong, fixed: https://tinyurl.com/athens-mac-m1

**Comment by sboesen** [comment]
Hello! Just checking if anything remains for this issue, looks like the gitcoin [bounty](https://gitcoin.co/issue/athensresearch/athens/749/100024993) is still open. Thanks!

**Comment by tangjeff0** [comment]
@sboesen not sure if this is still open!

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__The funding of 300.0 USD  (299.91 USD @ $1.0/USD) attached to this issue has been **cancelled** by the bounty submitter__

 
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.com/invite/83BK5z55yB/'>Gitcoin's Discord</a>
* $1,999,017.75 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
